module/Company/src/Company/Mapper/Label.php

```diff
@@ -0,0 +1,155 @@
+     */
+    public function findLabel($labelSlug)
+    {
+        return $this->getRepository()->findOneBy(['slug' => $labelSlug]);
+    }
+
+    public function findVisibleLabelByLanguage($labelLanguage)
+    {
+        $objectRepository = $this->getRepository(); // From clause is integrated in this statement
+        $qb = $objectRepository->createQueryBuilder('c');
+        $qb->select('c')->where('c.language=:lang');
+        $qb->andWhere('c.hidden=:hidden');
+        $qb->setParameter('lang', $labelLanguage);
+        $qb->setParameter('hidden', false);
+        $labels = $qb->getQuery()->getResult();
+
+        return $labels;
+    }
+
+    public function createNullLabel($lang, $translator)
+    {
+        $labelForJobsWithoutLabel =  new LabelModel();
+        $labelForJobsWithoutLabel->setHidden(false);
+        $labelForJobsWithoutLabel->setLanguageNeutralId(null);
+        $labelForJobsWithoutLabel->setLanguage($lang);
+        $labelForJobsWithoutLabel->setSlug("jobs");
+        $labelForJobsWithoutLabel->setName($translator->translate("Job"));
+
+        return $labelForJobsWithoutLabel;
+    }
+
+    /**
+     * Find the same label, but in the given language
+     *
+     */
+    public function siblingLabel($label, $lang)
+    {
+        $objectRepository = $this->getRepository(); // From clause is integrated in this statement
+        $qb = $objectRepository->createQueryBuilder('c');
+        $qb->select('c')->where('c.languageNeutralId=:labelID')->andWhere('c.language=:language');
+        $qb->setParameter('labelID', $label->getLanguageNeutralId());
+        $qb->setParameter('language', $lang);
+        $labels = $qb->getQuery()->getResult();
+        return $labels[0];
+    }
+    /**
+     * Inserts a new package into the given company
+     *
+     */
+    public function insert($lang, $id, $label = null)
+    {
+        if ($label == null) {
+            $label = new LabelModel();
+        }
+        $label->setLanguage($lang);
+        $label->setLanguageNeutralId($id);
+        $label->setHidden(false);
+        $this->em->persist($label);
+        $this->em->flush();
+        if ($id == -1) {
+            $id = $label->getId();
+        }
+        $label->setLanguageNeutralId($id);
+
+        return $label;
+    }
+
+    public function findAllLabelsById($labelId)
+    {
+        $objectRepository = $this->getRepository(); // From clause is integrated in this statement
+        $qb = $objectRepository->createQueryBuilder('c');
+        $qb->select('c')->where('c.languageNeutralId=:labelID');
+        $qb->setParameter('labelID', $labelId);
+        $labels = $qb->getQuery()->getResult();
+
+        return $labels;
+    }
+    /**
+     * Deletes the given label
+     *
+     */
+    public function delete($labelID)
+    {
+        $label = $this->findEditableLabel($labelID);
+        if (is_null($label)) {
+            return;
+        }
+
+        $this->em->remove($label);
+        $this->em->flush();
+    }
```

Comment:
`delete` or `remove` methods should only have an argument by object. You could create a `removeById` method, but it should simply be a wrapper around the normal `remove` method.

Reply:
I see your point, but this requires some slightly larger changes to the module. To be consistent with the remaining code within the module I chose not to change this.

Reply:
I would prefer that you fix these mistakes here at least. And possibly add an issue to refactor the rest of the module. Instead of making the refactoring an even more herculean task to be done later.

Reply:
While making these changes, I actually noticed this code is used in no place.